src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1574,6 +1590,45 @@ bool OutputStateMachineEngine::_GetOscSetColorTable(const std::wstring_view stri
     return success;
 }
 
+// Routine Description:
+// - Given a hyperlink string, attempts to parse the URI encoded. An 'id' parameter
+//   may be provided.
+//   If there is a URI, the well formatted string looks like:
+//          "<params>;<URI>"
+//   If there is no URI, we need to close the hyperlink and the string looks like:
+//          ";"
+// Arguments:
+// - string - the string containing the parameters and URI
+// - params - where to store the parameters
+// - uri - where to store the uri
+// Return Value:
+// - True if a URI was successfully parsed or if we are meant to close a hyperlink
+bool OutputStateMachineEngine::_ParseHyperlink(const std::wstring_view string,
+                                               std::wstring& params,
+                                               std::wstring& uri) const
+{
+    params.clear();
+    uri.clear();
+    const std::wstring idStr = L"id=";

```

Comment:
```suggestion
    static constexpr std::wstring_view hyperlinkIDParameter{ L"id=" };
```

then move it to somewhere near the top of the file with the other constants :smile:

Reply:
(otherwise, we construct a new string that contains "id=" every time we run into a hyperlink, when we really don't need to do that!)